src/software/new_geom/vector_test.cpp

```diff
@@ -0,0 +1,240 @@
+}
+
+TEST(VectorLogicTests, vector_proj_test)
+{
+    Vector u    = Vector(1, 2);
+    Vector v    = Vector(3, 4);
+    Vector proj = u.project(v);
+    EXPECT_EQ(1.32, proj.x());
+    EXPECT_EQ(1.76, proj.y());
+}
+
+TEST(VectorLogicTests, vector_dot_test)
+{
+    Vector u = Vector(-4, -9);
+    Vector v = Vector(-1, 2);
+    EXPECT_DOUBLE_EQ(-14, u.dot(v));
+}
+
+TEST(VectorLogicTests, vector_cross_product_test)
+{
+    Vector u = Vector(-4, -9);
+    Vector v = Vector(-1, 2);
+    EXPECT_DOUBLE_EQ(-17, u.cross(v));
+}
+
+TEST(VectorLogicTests, vector_orientation_test)
+{
+    Vector u = Vector(sqrt(2) / 2, sqrt(2) / 2);
+    EXPECT_EQ(Angle::ofDegrees(45), u.orientation());
+    Vector v = Vector(0, -1);
+    EXPECT_EQ(Angle::ofDegrees(-90), v.orientation());
+}
+
+TEST(VectorLogicTests, vector_is_nan_test)
+{
+    Vector v = Vector();
+    EXPECT_FALSE(v.isnan());
+    v.setX(NAN);
+    EXPECT_TRUE(v.isnan());
+    v = Vector();
+    EXPECT_FALSE(v.isnan());
+    v.setY(NAN);
+    EXPECT_TRUE(v.isnan());
+}
+
+TEST(VectorOperatorTests, vector_assignment_test)
+{
+    Vector u = Vector(2, 3);
+    Vector v = Vector();
+    v        = u;
+    EXPECT_EQ(2, v.x());
+    EXPECT_EQ(3, v.y());
+}
+
+TEST(VectorOperatorTests, vector_sum_test)
+{
+    Vector u = Vector(2, 3);
+    Vector v = Vector(-3, 5);
+    Vector w = u + v;
+    EXPECT_EQ(-1, w.x());
+    EXPECT_EQ(8, w.y());
+}
+
+TEST(VectorOperatorTests, vector_sum_set_test)
+{
+    Vector u = Vector(2, 3);
+    Vector v = Vector(-3, 5);
+    u += v;
+    EXPECT_EQ(-1, u.x());
+    EXPECT_EQ(8, u.y());
+}
+
+TEST(VectorOperatorTests, negate_vector_test)
+{
+    Vector u = Vector(3, -5);
+    Vector v = -u;
+    EXPECT_EQ(-3, v.x());
+    EXPECT_EQ(5, v.y());
+}
+
+TEST(VectorOperatorTests, vector_subtraction_test)
+{
+    Vector u = Vector(2, 3);
+    Vector v = Vector(-3, 5);
+    Vector w = u - v;
+    EXPECT_EQ(5, w.x());
+    EXPECT_EQ(-2, w.y());
+}
+
+TEST(VectorOperatorTests, vector_subtraction_set_test)
+{
+    Vector u = Vector(2, 3);
+    Vector v = Vector(-3, 5);
+    u -= v;
+    EXPECT_EQ(5, u.x());
+    EXPECT_EQ(-2, u.y());
+}
+
+TEST(VectorOperatorTests, scale_vector_test)
+{
+    Vector u = Vector(2, -3);
+    Vector w = u * 3;
+    EXPECT_EQ(6, w.x());
+    EXPECT_EQ(-9, w.y());
+}
+
+TEST(VectorOperatorTests, scale_set_vector_test)
+{
+    Vector u = Vector(2, -3);
+    u *= 3;
+    EXPECT_EQ(6, u.x());
+    EXPECT_EQ(-9, u.y());
+}
+
+TEST(VectorOperatorTests, divide_vector_test)
+{
+    Vector u = Vector(12, -3);
+    Vector w = u / 3;
+    EXPECT_EQ(4, w.x());
+    EXPECT_EQ(-1, w.y());
+}
+
+TEST(VectorOperatorTests, divide_set_vector_test)
+{
+    Vector u = Vector(12, -3);
+    u /= 3;
+    EXPECT_EQ(4, u.x());
+    EXPECT_EQ(-1, u.y());
+}
+
+TEST(VectorOperatorTests, vector_equality_inequality_test)
+{
+    Vector u = Vector();
+    Vector v = Vector(0, 0);
+    EXPECT_EQ(u, v);
+
+    v.setX(3);
+
+    EXPECT_FALSE(u == v);
+    EXPECT_TRUE(u != v);
+}
```

Comment:
Are we testing everything here, including all the operators?

Reply:
Yep every constructor/function/operator should be tested